components/callbacks/fx.go

```diff
@@ -48,9 +50,14 @@ var Module = fx.Module(
 
 func HTTPCallerProviderProvider(
 	clusterMetadata cluster.Metadata,
-	httpClientCache *cluster.FrontendHTTPClientCache,
+	rpcFactory common.RPCFactory,
 	logger log.Logger,
-) HTTPCallerProvider {
+) (HTTPCallerProvider, error) {
+	internalClient, err := rpcFactory.CreateLocalFrontendHTTPClient()
+	if err != nil {
+		return nil, fmt.Errorf("cannot create local frontend HTTP client: %w", err)
```

Comment:
If we can't get an internal client, wouldn't we still want to use the default client since the request may be external?

Reply:
This should never happen, it's either a bug or a misconfigured server. I think failing fast is the right approach here. I prefer this over silently doing the wrong thing.